pkg/gateway/gateway_test.go

```diff
@@ -4,11 +4,177 @@ package gateway
 // Licensed under the Apache License 2.0.
 
 import (
+	"context"
+	"errors"
 	"testing"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/mock/gomock"
+	"github.com/pires/go-proxyproto"
+	"github.com/sirupsen/logrus"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	mock_database "github.com/Azure/ARO-RP/pkg/util/mocks/database"
+	mock_env "github.com/Azure/ARO-RP/pkg/util/mocks/env"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+	utilnet "github.com/Azure/ARO-RP/pkg/util/net"
 )
 
+func TestNewGatway(t *testing.T) {
+	emptyEnv := &azureclient.AROEnvironment{
+		Environment: azure.Environment{
+			ActiveDirectoryEndpoint:    "",
+			ResourceManagerEndpoint:    "",
+			ContainerRegistryDNSSuffix: "dnsSuffix",
+		},
+	}
+
+	for _, tt := range []struct {
+		name              string
+		gatewayDomains    string
+		acrResourceID     string
+		mocks             func(env *mock_env.MockCore)
+		expectedAllowList []string
+		wantErr           error
+	}{
+		{
+			name: "Empty gatewayDomains, empty Environment, no acrResourceID so the allowList is empty",
```

Comment:
this would be updated per my previous comment, we'll want an err because our gateway is missing important domains